File: doc-testing/src/sdk_openai_simple.ts
import "dotenv/config";
import { init, wrapOpenAI, flush, shutdown } from "neatlogs";
import OpenAI from "openai";

async function main() {
  await init({
    apiKey: process.env.NEATLOGS_API_KEY,
    endpoint: process.env.NEATLOGS_ENDPOINT,
    workflowName: "openai-demo",
  });

  const client = wrapOpenAI(new OpenAI());

  const res = await client.chat.completions.create({
    model: "gpt-4o",
    messages: [{ role: "user", content: "In one sentence, what is OpenAI?" }],
  });
  console.log(res.choices[0].message.content);

  await flush();
  await shutdown();
}

main().catch(console.error);

File: agents/src/agents/sdk_openai_simple.py
import os

from dotenv import load_dotenv
import neatlogs
from openai import OpenAI

load_dotenv()

neatlogs.init(api_key=os.environ["NEATLOGS_API_KEY"], endpoint=os.environ.get("NEATLOGS_ENDPOINT", "http://localhost:4100"), workflow_name="openai-demo")

client = neatlogs.wrap(OpenAI())

resp = client.chat.completions.create(
    model="gpt-4o",
    messages=[{"role": "user", "content": "In one sentence, what is OpenAI?"}],
)
print(resp.choices[0].message.content)

neatlogs.flush()
neatlogs.shutdown()
